blocks/address/src/editable-field.tsx

```diff
@@ -0,0 +1,103 @@
+import { faPenToSquare } from "@fortawesome/free-regular-svg-icons";
+import { FontAwesomeIcon } from "@hashintel/design-system";
+import {
+  Box,
+  Fade,
+  IconButton,
+  inputBaseClasses,
+  TextField,
+  TextFieldProps,
+  useTheme,
+} from "@mui/material";
+import { useRef, useState } from "react";
+
+export const EditableField = ({
+  iconSize,
+  readonly,
+  value,
+  placeholder,
+  sx,
+  onBlur,
+  ...props
+}: { iconSize: string; readonly?: boolean } & TextFieldProps) => {
+  const { transitions } = useTheme();
+
+  const [hovered, setHovered] = useState(false);
+  const [editing, setEditing] = useState(!readonly && !value ? true : false);
+  const inputRef = useRef<HTMLDivElement | null>(null);
+
+  return (
+    <Box
+      onMouseEnter={() => setHovered(true)}
+      onMouseLeave={() => setHovered(false)}
+      sx={{
+        display: "flex",
+      }}
+    >
+      <TextField
+        {...props}
+        value={value}
+        autoFocus
+        onBlur={(event) => {
+          setEditing(false);
+          onBlur?.(event);
+        }}
+        onKeyDown={({ code }) => {
+          if (code === "Enter") {
+            setEditing(false);
+          }
+        }}
+        multiline
+        inputRef={inputRef}
+        inputProps={{
+          readOnly: !editing,
+          ...props.inputProps,
+        }}
+        variant="standard"
+        placeholder={!readonly ? placeholder : undefined}
+        sx={{
+          width: "100%",
+          [`.${inputBaseClasses.root}`]: {
+            paddingTop: 0,
+            transition: transitions.create("padding"),
+            ...(!editing ? { p: 0 } : {}),
+          },
+          ...(!editing
+            ? {
+                "& ::before": {
+                  borderWidth: "0 !important",
+                },
+                "& ::after": {
+                  borderWidth: "0 !important",
+                },
+              }
+            : {}),
+          ...(Array.isArray(sx) ? sx : [sx]),
+        }}
+      />
+
+      {!readonly ? (
+        <Fade in={hovered && !editing}>
+          <Box sx={{ position: "relative" }}>
+            <IconButton
+              onClick={() => {
+                setEditing(!editing);
```

Comment:
Shouldn't this be `true`? This is to enable editing, not to toggle it.

if it's a toggle it should only focus the input if editing is being set to `true`

Reply:
This button was a toggle in its first iteration, forgot to change it. Addressed in 39eeca67e607adb15706c7258e6fa85d5b6c6277